### DISCUSSION TITLE
Commit not all modifications of one file

### DISCUSSION THREAD

**Discussion by andr1972** [libgit2/libgit2#6390] [Q&A]
New version of file to commit has N modifications. Is possible with libgit2 choose which modifications will committed? 
For example, file has 5  correct modifications and one modification is temporary helper for debugging, not to commit.
CLion has these ability.

**Comment by ethomson** [comment]
We don't provide a mechanism like `git apply -p`, but you can choose whatever you want to stage by adding it to the index. You'll need to create the file that you want to commit (in other words, choose the modifications) yourself, though.
